internal/cachecontroller/cache_controller.go

```diff
@@ -171,19 +173,28 @@ func (c *InMemoryCacheController) findChangesAndInvalidate(ctx context.Context,
 		return
 	}
 
+	timestampOfLastIteratorInvalidation := time.Now().Add(-c.iteratorCacheTTL)
```

Comment:
Discussed offline - we also want to use timestampOfLastIteratorInvalidation to check for `entry.LastModified.Before` above.  This will provide higher consistency